plan/client.go

```diff
@@ -29,23 +28,7 @@ func New(params *stripe.PlanParams) (*stripe.Plan, error) {
 
 func (c Client) New(params *stripe.PlanParams) (*stripe.Plan, error) {
 	body := &stripe.RequestValues{}
-	body.Add("id", params.ID)
-	body.Add("name", params.Name)
-	body.Add("amount", strconv.FormatUint(params.Amount, 10))
-	body.Add("currency", string(params.Currency))
-	body.Add("interval", string(params.Interval))
-
-	if params.IntervalCount > 0 {
-		body.Add("interval_count", strconv.FormatUint(params.IntervalCount, 10))
-	}
-
-	if params.TrialPeriod > 0 {
-		body.Add("trial_period_days", strconv.FormatUint(params.TrialPeriod, 10))
-	}
-
-	if len(params.Statement) > 0 {
-		body.Add("statement_descriptor", params.Statement)
-	}
+	params.Params.AppendTo(body)
```

Comment:
Lets move away from constructing request bodies from clients and into `AppendTo` functions implemented by parameter structs. This way we can test the logic much more safely (and it's also reusable which is another big advantage).